tests/phpunit/includes/testcase-multisites.php

```diff
@@ -0,0 +1,312 @@
+<?php
+
+/**
+ * Test case offering a standardized way to test blogs in multisite.
+ */
+abstract class PLL_Multisites_TestCase extends WP_UnitTestCase {
+	/**
+	 * Blog in plain permalinks without Polylang.
+	 *
+	 * @var WP_Site
+	 */
+	protected static $blog_without_pll_plain_links;
+
+	/**
+	 * Blog in pretty permalinks without Polylang.
+	 *
+	 * @var WP_Site
+	 */
+	protected static $blog_without_pll_pretty_links;
+
+	/**
+	 * Blog in plain permalinks with Polylang (i.e. default installation).
+	 * Created languages are English and French.
+	 *
+	 * @var WP_Site
+	 */
+	protected static $blog_with_pll_default_links;
+
+	/**
+	 * Blog in pretty permalinks with Polylang and language as directory.
+	 * Created languages are English and French.
+	 *
+	 * @var WP_Site
+	 */
+	protected static $blog_with_pll_directory; // Main blog.
+
+	/**
+	 * Blog in pretty permalinks with Polylang and language as domains.
+	 * Created languages are English and German.
+	 *
+	 * @var WP_Site
+	 */
+	protected static $blog_with_pll_domains;
+
+	/**
+	 * Pretty permalinks structure.
+	 *
+	 * @var string
+	 */
+	protected $pretty_structure = '/%postname%/';
+
+	/**
+	 * Plain permalinks structure.
+	 *
+	 * @var string
+	 */
+	protected $plain_structure = '';
+
+	/**
+	 * Languages data for their creation keyed by language slug.
+	 *
+	 * @var array
+	 */
+	protected $languages = array(
+		'en' => array(
+			'name'       => 'English',
+			'slug'       => 'en',
+			'locale'     => 'en_US',
+			'rtl'        => 0,
+			'flag'       => 'us',
+			'term_group' => 0,
+		),
+		'fr' => array(
+			'name'       => 'Français',
+			'slug'       => 'fr',
+			'locale'     => 'fr_FR',
+			'rtl'        => 0,
+			'flag'       => 'fr',
+			'term_group' => 1,
+		),
+		'de' => array(
+			'name'       => 'Deutsch',
+			'slug'       => 'de',
+			'locale'     => 'de_DE',
+			'rtl'        => 0,
+			'flag'       => 'de',
+			'term_group' => 2,
+		),
+	);
+
+	public function set_up() {
+		parent::set_up();
+
+		// Create all sites.
+		$factory = $this->factory();
+
+		self::$blog_with_pll_directory = get_site( 1 );
+
+		self::$blog_with_pll_domains = $factory->blog->create_and_get(
+			array(
+				'domain' => 'polylang-domains.org',
+			)
+		);
+
+		self::$blog_with_pll_default_links = $factory->blog->create_and_get(
+			array(
+				'domain' => 'polylang-plain.org',
+			)
+		);
+
+		self::$blog_without_pll_plain_links = $factory->blog->create_and_get(
+			array(
+				'domain' => 'fake.com',
+			)
+		);
+
+		self::$blog_without_pll_pretty_links = $factory->blog->create_and_get(
+			array(
+				'domain' => 'wordpress.org',
+			)
+		);
```

Comment:
To improve readability in case we use these domains in tests:
```suggestion
		self::$blog_without_pll_plain_links = $factory->blog->create_and_get(
			array(
				'domain' => 'plain.org',
			)
		);

		self::$blog_without_pll_pretty_links = $factory->blog->create_and_get(
			array(
				'domain' => 'pretty.org',
			)
		);
```

Reply:
Changed in [5732659](https://github.com/polylang/polylang/pull/1352/commits/57326590f4a2234c7f9e06f98a9e7c32b64ab781).